Dependency zip.rs v4.3.0 requires a MSRV of 1.85.0

Zip.rs v4.3.0 requires a MSRV of v1.85.0 whereas Calamine's stated MSRV is v1.73.0. This is causing a failure in CI and presumably for any users with rustc < v1.85.0:



```
Run cargo build
    Updating crates.io index
 Downloading crates ...
  ...
error: package `zip v4.3.0` cannot be built because it requires rustc 1.82.0 or newer, 
while the currently active rustc version is 1.75.0
Either upgrade to rustc 1.82.0 or newer, or use
cargo update zip@4.3.0 --precise ver
where `ver` is the latest version of `zip` supporting rustc 1.75.0
Error: Process completed with exit code 101.
```


https://github.com/tafia/calamine/actions/runs/16188594372/job/45698992946

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,7 +22,7 @@ fast-float2 = "0.2"
 log = "0.4"
 serde = "1.0"
 quick-xml = { version = "0.37", features = ["encoding"] }
-zip = { version = "4.0", default-features = false, features = ["deflate"] }
+zip = { version = "~4.2.0", default-features = false, features = ["deflate"] }
 chrono = { version = "0.4", features = [
     "serde",
 ], optional = true, default-features = false }

